Add Compass documentation to Kyma (#5306)
kyma-project/kyma: docs/compass/02-01-compass.md

---
title: Basic architecture
type: Architecture
---

Compass is a central place which stores applications and runtimes configurations, and then propagates that information accordingly. It also plays a crucial role in establishing a trusted connection between applications and runtimes. The basic workflow looks as follows:

![Basic architecture](./assets/architecture.svg)

1. Administrator adds runtimes and applications, and configures them using Compass GraphQL API.

> 2. Agent, a component that is integrated in every Kyma runtime, continuously fetches the actual configuration from Compass. It also sends information about runtime health checks to Compass.

Reviewer: health checks are not yet implemented, should we mention them in docu?

Author: Agreed with PK that we'll use the future tense here.

> 3. In case an application has optional webhooks configured, Compass notifies an application about any Events that concern a given application. _tutaj_

Reviewer: what _tutaj_ ?

Author: oops, my bad :D

Compass does not participate in any business flow. After establishing a trusted connection between an application and a runtime, they communicate directly with each other.

## Scenarios

In order to connect and group your applications and runtimes, assign them to the same scenario.

> A scenario is a simple label with the **scenarios** key. If an application or a runtime is not explicitly assigned to any scenario, they belong to the `default` one. Applications and runtimes are automatically removed from the `default` scenario after you assign them to any other scenario. You can assign applications and runtimes to many scenarios. See the example:

Reviewer: Runtimes are not assigned to default scenario AFAIK.

Author: That's true, I've fixed that sentence.

> A scenario is a simple label with the **scenarios** key. If an application or a runtime is not explicitly assigned to any scenario, they belong to the `default` one. Applications and runtimes are automatically removed from the `default` scenario after you assign them to any other scenario. You can assign applications and runtimes to many scenarios. See the example:

Reviewer: to check if this is true:
`Applications and runtimes are automatically removed from the `default` scenario after you assign them to any other scenario`

Author: This is how it should work 👀

![Scenarios](./assets/scenarios.svg)

> `Application 2` belongs to the `production` and `stage` scenarios, which means that both `Runtime 1` and `Runtime 2` can consume its API or Events. Communication between components that do not belong to the same scenario is forbidden. For example, `Application 3` cannot communicate with `Runtime 1`.

Reviewer: some time ago we said that `production` and `stage` scenarios can be misleading, maybe we can use sth like: "black-friday-campaign", "christmas-campaign", "summer-campaign"

Author: Are you kidding me? 😄

Author: Agreed with PK that we'll name them `campaing`, `marketing`, and `commerce`.

---
title: Basic architecture
type: Architecture
---

> Compass is a central place which stores applications and Runtimes configurations, and then propagates that information accordingly. It also plays a crucial role in establishing a trusted connection between applications and Runtimes. The basic workflow looks as follows:

Reviewer 2: ```suggestion
Compass is a central place which stores application and Runtime configuration and then propagates that information accordingly. It also plays a crucial role in establishing a trusted connection between applications and Runtimes. The basic workflow looks as follows:
```

Reviewer 2: `propagates that information accordingly` maybe we can explain it a bit or at least add where it propagates that info?

![Basic architecture](./assets/architecture.svg)

> 

Author: Good point, thanks!

1. Administrator adds Runtimes and applications, and configures them using Compass.

> 2. Agent, a component that is integrated in every Kyma Runtime, continuously fetches the actual configuration from Compass. In the future releases, Agent will also send information about Runtime health checks to Compass.

Reviewer 2: ```suggestion
2. Agent, a component that is integrated in every Kyma Runtime, continuously fetches the latest configuration from Compass. In the future releases, Agent will also send information about Runtime health checks to Compass.
```

Reviewer 2: unless you actually meant "actually" here ;)

> 3. In case an application has optional webhooks configured, Compass notifies an application about any Events that concern a given application.

Reviewer 2: ```suggestion
3. If an application has optional webhooks configured, Compass notifies an application about any Events that concern a given application.
```

Compass does not participate in any business flow. After establishing a trusted connection between an application and a Runtime, they communicate directly with each other.

## Scenarios

In order to connect and group your applications and Runtimes, assign them to the same scenario.

> A scenario is a simple label with the **scenarios** key. If an application is not explicitly assigned to any scenario, it belongs to the `default` one. The application is automatically removed from the `default` scenario after you assign it to any other scenario. Runtimes by default are not assigned to any scenario. You can assign applications and Runtimes to many scenarios. See the example:

Reviewer 2: ```suggestion
A scenario is a simple label with the **scenarios** key. If an application is not explicitly assigned to any scenario, it belongs to the `default` one. The application is automatically removed from the `default` scenario after you assign it to any other scenario. By default, Runtimes are not assigned to any scenario. You can assign applications and Runtimes to multiple scenarios. See the example:
```

![Scenarios](./assets/scenarios.svg)

>**CAUTION:** Currently, you can connect an application only to one Runtime at the same time.  

> `Application 2` belongs to the `campaign` and `marketing` scenarios, which means that both `Runtime 1` and `Runtime 2` can consume its APIs or Events, but not at the same time. If you try to connect a Runtime to an application that is already assigned to another Runtime, the communication fails. Communication between components that do not belong to the same scenario is completely forbidden. For example, `Application 3` cannot communicate with `Runtime 1`.

Reviewer 2: ```suggestion
`Application 2` belongs to the `campaign` and `marketing` scenarios, which means that both `Runtime 1` and `Runtime 2` can consume its APIs or Events, but not at the same time. If you try to connect a Runtime to an application that is already assigned to another Runtime, the communication fails. Communication between components that do not belong to the same scenario is not possible. For example, `Application 3` cannot communicate with `Runtime 1`.
```
